src/app/services/signalr/signalr.service.ts

```diff
@@ -108,13 +118,23 @@ export class SignalRService {
 
   private addHandlers() {
     this.addUserHandlers();
+    this.addCurrentUsersHandlers();
   }
 
-  private addUserHandlers() {
-    this.hubConnection.on('ActiveVirtualMachine', (vmId: string) => {
+  public addUserHandlers() {
+    this.hubConnection.on('ActiveVirtualMachine', (vmId: string, userId: string) => {
       this.vmService.setActive(vmId);
     });
   }
+
+  public addCurrentUsersHandlers() {
+    if (this.vmId) {
```

Comment:
I think this if may be unnecessary since the client shouldn't receive these broadcasts if they are not joined to the group anymore anyway.